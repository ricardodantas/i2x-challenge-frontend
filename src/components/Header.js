import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class Header extends Component {
  static propTypes = {
    children: PropTypes.any.isRequired,
  }

  render() {
    const { children } = this.props

    return (
      <header>
        {children}
      </header>
    )
  }
}
