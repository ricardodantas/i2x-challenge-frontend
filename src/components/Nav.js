import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Link, Redirect} from 'react-router-dom'
import { Container } from 'semantic-ui-react'
import {logout} from '../actions/login'

class Nav extends Component {

  static propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    logout: PropTypes.func.isRequired,
  }

  logout() {
    const {dispatch, logout} = this.props

    dispatch(logout())
  }

  render() {
    const {isAuthenticated} = this.props

    if(!isAuthenticated) {
      return null
    }

    return (
      <div className="ui attached stackable menu">
        <Container>
          <Link to="/" className="active item">Home Page</Link>
          <div className="right menu">
            <a className="ui item" onClick={this.logout.bind(this)}>
              Logout
            </a>
          </div>
        </Container>
      </div>
    )
  }
}


const mapDispatchToProps = (dispatch) => {
  return {
    dispatch,
    logout
  }
}

const mapStateToProps = (state) => {
  return {}
}

export default connect(mapStateToProps, mapDispatchToProps)(Nav)
