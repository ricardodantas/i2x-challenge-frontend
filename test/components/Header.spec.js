import React from 'react'
import { spy } from 'sinon'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Header from '../../src/components/Header'
import Nav from '../../src/components/Nav'

describe('<Header />', () => {

  it('should render', () => {
    const tree = renderer.create(
      <Header><div></div></Header>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

})
