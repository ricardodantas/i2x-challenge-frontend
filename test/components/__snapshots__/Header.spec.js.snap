// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render 1`] = `
<header>
  <div />
</header>
`;
